kakarot_scripts/utils/kakarot.py

```diff
@@ -747,29 +745,97 @@ async def store_bytecode(bytecode: Union[str, bytes], **kwargs):
     return evm_address
 
 
-async def deploy_with_presigned_tx(
-    deployer_evm_address: str, signed_tx: bytes, amount=0.1, name="", max_fee=None
-):
-    deployer_starknet_address = await deploy_and_fund_evm_address(
-        deployer_evm_address, amount
+# async def deploy_with_presigned_tx(
+#     deployer_evm_address: str, signed_tx: bytes, amount=0.1, name="", max_fee=None
+# ):
+#     deployer_starknet_address = await deploy_and_fund_evm_address(
+#         deployer_evm_address, amount
+#     )
+#     nonce = (
+#         await _call_starknet(
+#             "account_contract", "get_nonce", address=deployer_starknet_address
+#         )
+#     ).nonce
+#     if nonce != 0:
+#         logger.info(
+#             f"ℹ️  Nonce for {deployer_evm_address} is not 0 ({nonce}), skipping transaction"
+#         )
+#         return
+
+#     response = await send_pre_eip155_transaction(
+#         deployer_evm_address, deployer_starknet_address, signed_tx, max_fee
+#     )
+
+#     receipt, response, success, gas_used = response
+#     deployed_address = response[1]
+#     logger.info(f"✅ {name} Deployed at: 0x{deployed_address:040x}")
+#     deployed_starknet_address = await get_starknet_address(deployed_address)
+#     return {"address": deployed_address, "starknet_address": deployed_starknet_address}
+
+
+async def whitelist_pre_eip155_tx(name: str) -> bool:
+    tx_instance = PRE_EIP155_TX[name]
+    deployer_evm_address = tx_instance["deployer"]
+    tx_instance["address"]
+    signed_tx = tx_instance["signed_tx"]
+    amount = tx_instance["required_eth"]
+
+    # check the nonce of the deployer for an early return if it's not 0.
+    # Either the nonce is 0, or the account should not be deployed already.
+    deployer_starknet_address = (
+        f"0x{await get_starknet_address(deployer_evm_address):064x}"
     )
-    nonce = (
-        await _call_starknet(
-            "account_contract", "get_nonce", address=deployer_starknet_address
-        )
-    ).nonce
+    kakarot_contract = _get_starknet_contract("kakarot")
+    try:
+        nonce = (
+            await kakarot_contract.functions["eth_get_transaction_count"].call(
+                int(deployer_evm_address, 16)
+            )
+        ).tx_count
+    except Exception as e:
+        if (
+            f"Requested contract address {deployer_starknet_address} is not deployed"
+            in str(e.data)
+        ):
+            nonce = 0
+        else:
+            raise e
     if nonce != 0:
         logger.info(
             f"ℹ️  Nonce for {deployer_evm_address} is not 0 ({nonce}), skipping transaction"
         )
         return
 
-    response = await send_pre_eip155_transaction(
-        deployer_evm_address, deployer_starknet_address, signed_tx, max_fee
+    # Deploy and fund deployer for the authorization callback
+    await deploy_and_fund_evm_address(deployer_evm_address, amount)
```

Comment:
i don't think that deploying the deployer belongs to the whitelist process, move this to `send_pre_eip155_transaction`

Reply:
I need to deploy the deployer to call `set_authorized_pre_eip155_tx` on it.